crates/sui-graphql-rpc/src/types/query.rs

```diff
@@ -239,3 +350,89 @@ impl Query {
             .extend()
     }
 }
+
+// Intermediate enum to handle the two possible types of `tx_bytes` in `dry_run_transaction_block` usiing FN
+enum DryRunExecArg {
+    TransactionData(TransactionData),
+    TransactionKindWithMeta {
+        kind: TransactionKind,
+        sender: NativeSuiAddress,
+        gas_price: Option<u64>,
+        epoch: Option<u64>,
+        gas_objects: Option<Vec<NativeSuiAddress>>,
+    },
+}
+
+impl DryRunExecArg {
+    pub fn into_transaction_data(self) -> TransactionData {
+        match self {
+            DryRunExecArg::TransactionData(td) => return td,
+
+            // We hit this scenario if the TransactionKind and meta are provided
+            // and skip_checks is false.
+            // This requires forming a full TransactionData object.
+            DryRunExecArg::TransactionKindWithMeta {
+                kind,
+                sender,
+                gas_price,
+                epoch,
+                gas_objects,
+            } => {
+                // TODO: We need to handle the gas data
+
+                // We dont have enough info for these
+                // let gas_payment: ObjectRef;
+                // let gas_budget: u64;
+                TransactionData::new_with_epoch(
+                    kind,
+                    sender,
+                    gas_payment,
+                    gas_budget,
+                    gas_price,
+                    epoch,
+                )
+            }
+        }
+    }
+
+    pub fn into_transaction_kind_with_meta(
+        self,
+    ) -> (
+        TransactionKind,
+        /* Sender */ NativeSuiAddress,
+        /* Gas price */ Option<u64>,
+        /* Epoch */ Option<u64>,
+    ) {
+        match self {
+            DryRunExecArg::TransactionData(td) => {
+                let kind = td.kind();
+                let sender = td.sender().into();
+                let gas_price = td.gas_price();
+                let epoch = td.expiration().into();
+                (kind.clone(), sender, Some(gas_price), epoch)
+            }
+            DryRunExecArg::TransactionKindWithMeta {
+                kind,
+                sender,
+                gas_price,
+                epoch,
+                ..
+            } => (kind, sender, gas_price, epoch),
+        }
+    }
+}
+
+// Intermediate enum to handle the two possible return types of `dry_run_transaction_block` using FN
+enum DryRunExecResult {
+    DryRun(DryRunTransactionBlockResponse),
+    DevInspect(DevInspectResults),
+}
+
+impl DryRunExecResult {
+    pub fn into_dry_run_result(self) -> Result<DryRunResult> {
+        match self {
```

Comment:
Basically this portion here seems like we can shore horn some stuff but really wanna make sure we align on completeness